carnival/carnival.ts

```diff
@@ -0,0 +1,1017 @@
+        let vy = carnival.yComponent(parentBall.angle, parentBall.pow);
+        let ay = parentBall.gravity;
+        let ax = parentBall.wind;
+        let p = parentBall.pow;
+        if (!kind) { kind = SpriteKind.Projectile; }
+        return createProjectileBall(img, vx, vy, ax, ay, p, kind, parentBall);
+    }
+
+    /**
+     * Create a new sprite with given speed, and place it at the edge of the screen so it moves towards the middle.
+     * The sprite auto-destroys when it leaves the screen. You can modify position after it's created.
+     */
+    //% group="Ball"
+    //% color="#b70082"
+    //% blockId=spritescreateprojectileball block="ball $img=screen_image_picker vx $vx vy $vy of kind $kind=spritekind||based on $parentBall=variables_get(myBall)"
+    //% blockSetVariable=myBall
+    //% inlineInputMode=inline
+    //% expandableArgumentMode=toggle
+    //% help=github:carnival/docs/spritescreateprojectileball
+    function createProjectileBall(img: Image, vx: number, vy: number, ax: number, ay: number, power: number, kind?: number, parentBall?: Ball) {
+        const s = carnival.create(img, kind || SpriteKind.Projectile);
+        const sc = game.currentScene();
+
+        if (parentBall) {
+            s.setPosition(parentBall.x, parentBall.y);
+            s.vx = xComponent(parentBall.angle, parentBall.pow);
+            s.vy = carnival.yComponent(parentBall.angle, parentBall.pow);
+            s.ay = parentBall.gravity;
+            s.ax = parentBall.wind;
+            s.pow = parentBall.pow;
+        } else {
+            // put it at the edge of the screen so that it moves towards the middle
+            // If the scene has a tile map, place the sprite fully on the screen
+            const xOff = sc.tileMap ? -(s.width >> 1) : (s.width >> 1) - 1;
+            const yOff = sc.tileMap ? -(s.height >> 1) : (s.height >> 1) - 1;
+            const cam = game.currentScene().camera;
+
+            let initialX = cam.offsetX;
+            let initialY = cam.offsetY;
+
+            if (vx < 0) {
+                initialX += screen.width + xOff;
+            } else if (vx > 0) {
+                initialX += -xOff;
+            }
+
+            if (vy < 0) {
+                initialY += screen.height + yOff;
+            } else if (vy > 0) {
+                initialY += -yOff;
+            }
+
+            s.setPosition(initialX, initialY);
+        }
+
+        s.moon.destroy();
+        s.flags |= sprites.Flag.AutoDestroy | sprites.Flag.DestroyOnWall;
+
+        return s;
+    }
+
+
+
+    /**
+     * Convert degrees to radians
+     * @param degree to convert
+     * @return converted value in radians
+     */
+    function degreeToRadian(degree: number): number {
+        return degree * Math.PI / 180;
+    }
+
+    /**
+     * Evaluate the x component of a given vector
+     * @param degree angle of vector
+     * @param magnitude magnitude of vector
+     * @return x component of vector
+     */
+    //% color="#b70082"
```

Comment:
this color annotation won't do anything unless this function has a block annotation

Reply:
It's leftover from when I exported it.